src/Common/Commands.Common/AzureLongRunningJob.cs

```diff
@@ -243,13 +243,22 @@ public override string StatusMessage
                 returnValue.MyInvocation.BoundParameters.Add(parameter.Key, parameter.Value);
             }
 
-            foreach (var field in returnType.GetFields(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.DeclaredOnly))
+            CopyCmdletFields(returnType, cmdlet, returnValue);
+            cmdlet.SafeCopyParameterSet(returnValue);
+            return returnValue as U;
+        }
+
+        static void CopyCmdletFields<U>(Type targetType, U sourceCmdlet, U targetCmdlet) where U : AzurePSCmdlet
```

Comment:
@markcowl not sure if it's possible, but could an object of type `AzurePSCmdlet` be passed as `targetType` in this method? If so, is there any problem with copying the properties of the `PSCmdlet` type (which `AzurePSCmdlet` inherits from)?